modules/gapi/test/streaming/gapi_gstreamer_pipeline_facade_int_tests.cpp

```diff
@@ -0,0 +1,182 @@
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink1 "
+                       "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink2");
+
+    auto videotestsrcs = pipelineFacade.getElementsByFactoryName("videotestsrc");
+    EXPECT_EQ(2u, videotestsrcs.size());
+    for (auto&& src : videotestsrcs) {
+        EXPECT_TRUE(comparator(src, "videotestsrc"));
+    }
+
+    auto appsinks = pipelineFacade.getElementsByFactoryName("appsink");
+    EXPECT_EQ(2u, appsinks.size());
+    for (auto&& sink : appsinks) {
+        EXPECT_TRUE(comparator(sink, "appsink"));
+    }
+}
+
+TEST(GStreamerPipelineFacadeTest, GetElByNameUnitTest)
+{
+    auto comparator = [](GstElement* element, const std::string& elementName) {
+        cv::gapi::wip::gst::GStreamerPtr<gchar> name(gst_element_get_name(element));
+        return name && (0 == strcmp(name, elementName.c_str()));
+    };
+
+    cv::gapi::wip::gst::GStreamerPipelineFacade
+        pipelineFacade("videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink1 "
+                       "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink2");
+
+    auto appsink1 = pipelineFacade.getElementByName("sink1");
+    GAPI_Assert(appsink1 != nullptr);
+    EXPECT_TRUE(comparator(appsink1, "sink1"));
+    auto appsink2 = pipelineFacade.getElementByName("sink2");
+    GAPI_Assert(appsink2 != nullptr);
+    EXPECT_TRUE(comparator(appsink2, "sink2"));
+}
+
+TEST(GStreamerPipelineFacadeTest, CompletePrerollUnitTest)
+{
+    cv::gapi::wip::gst::GStreamerPipelineFacade
+        pipelineFacade("videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink1 "
+                       "videotestsrc is-live=true pattern=colors num-buffers=10 ! "
+                       "videorate ! videoscale ! "
+                       "video/x-raw,width=1920,height=1080,framerate=3/1 ! "
+                       "appsink name=sink2");
+
+    auto appsink = pipelineFacade.getElementByName("sink1");
+    pipelineFacade.completePreroll();
+
+    cv::gapi::wip::gst::GStreamerPtr<GstSample> prerollSample(
+    //        gst_app_sink_try_pull_preroll(GST_APP_SINK(appsink), 5 * GST_SECOND));
```

Comment:
Debug code?

Reply:
Thanks for a catch!! This code should be version-dependent!

Reply:
Fixed!